packages/components/src/number-control/index.tsx

```diff
@@ -60,42 +59,24 @@ function UnforwardedNumberControl(
 		spinControls = 'none';
 	}
 
+	if ( typeof valueProp === 'number' ) {
+		// TODO: deprecate `value` as a `number`
+	}
+
+	const valuePropAsString =
```

Comment:
Key change: from this line on, `NumberControl` treats the incoming `value` as a `string`